distribution/launcher/NOTICE

```diff
@@ -0,0 +1,112 @@
+Enso
```

Comment:
1. Who created this file?
2. What's the future recipe for updating it?
3. Is it documented?
4. Is this list exhaustive?
5. How do you know this is enough to satisfy these licenses?

Reply:
1 - I have added it by concatenating NOTICEs based on the licenses of the projects that we (may) currently include in the build.
2, 3 - I did miss that - I will add documentation specifying that it should be updated.
4. It should include all (transitive) dependencies that are used in the launcher (for now there aren't many).
5. I have spent a fair amount of time reading them, but indeed I am not sure and would rather get a confirmation from someone who has better knowledge of legal affairs.

Reply:
OK. Document what currently is here. After this PR lands, open a task for legal review of these, assign @sylwiabr and @iamrecursion

Reply:
I have documented the current process at https://github.com/enso-org/enso/blob/9a6f41e21932ebf3ef6eabc8b0c3bc21dcda4882/docs/distribution/licences.md

I will open the mentioned task after the PR is merged.